sdk/storage/storage-queue/src/Pipeline.ts

```diff
@@ -202,7 +206,10 @@ export function newPipeline(
     new BrowserPolicyFactory(),
     deserializationPolicy(), // Default deserializationPolicy is provided by protocol layer
     new RetryPolicyFactory(pipelineOptions.retryOptions),
-    new LoggingPolicyFactory()
+    logPolicy(
+      logger.info, {
+        allowedQueryParameters: ["api-version"]
```

Comment:
>allowedQueryParameters: ["api-version"] [](start = 8, length = 39)

Do we have blacklist?

Following info should be skipped during logging:
- `sig` parameter in the url
- `Authentication` header in the header values

Reply:
Anything not in the [default lists](https://github.com/Azure/azure-sdk-for-js/commit/d35927284640038bcf1f2d78135d95fbb46e965a) plus what is specified here is considered to be blacklisted, so the two you mentioned will show up as "REDACTED" in the logs.

Reply:
I noticed there are other values are shown as REDACTED, for example x-ms-content-crc64, etag.

For storage, number of whitelist values is far larger than blacklisted values. Maintaining a short list of blacklist is more maintainable than a long whitelist list. Every time storage releases new API version, there will be new headers. And as protocol layer is auto generated, manually check new added headers every time sounds not perfect.

Reply:
I believe the strategy we're going for across all SDK languages is to have an "allowed" list instead of a "deny" list just to avoid any unintended exposure of PII or secrets by default.  It does require more work on part of the service SDK teams to add new allowed items, but I think that's better than unintentionally causing sensitive data to be exposed when new service headers are added.

/cc @bterlson

Reply:
Avoiding unintentional PII exposure is the high order bit here, and the allow list is the way to do it - it's too risky to make sure we're always on top of new headers/query params so we can sanitize them when required. See the [general guidelines](https://azure.github.io/azure-sdk/general_implementation.html#general-logging-requests) for more info.